src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +266,54 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Add any pending ICE candidates from the queue for the client ID
+        Log.d(TAG, "Pending ice candidates found? " + pendingIceCandidatesMap.get(clientId));
+        Queue<IceCandidate> pendingIceCandidatesQueueByClientId = pendingIceCandidatesMap.get(clientId);
+        while(!pendingIceCandidatesQueueByClientId.isEmpty()) {
+            final IceCandidate iceCandidate = pendingIceCandidatesQueueByClientId.peek();
+            final PeerConnection peer = peerConnectionFoundMap.get(clientId);
+            final boolean addIce = peer.addIceCandidate(iceCandidate);
+            Log.d(TAG, "Added ice candidate after SDP exchange " + iceCandidate + " " + (addIce ? "Successfully" : "Failed"));
+            pendingIceCandidatesQueueByClientId.remove();
+        }
+        // After sending pending ICE candidates, the client ID's peer connection need not be tracked
+        pendingIceCandidatesMap.remove(clientId);
+    }
+
+    private void checkAndAddIceCandidate(Event message, IceCandidate iceCandidate) {
+        // if answer/offer is not received, it means peer connection is not found. Hold the received ICE candidates in the map.
+
+        if(!peerConnectionFoundMap.containsKey(message.getSenderClientId())) {
+            Log.d(TAG, "SDP exchange is not complete. Ice candidate " + iceCandidate + " + added to pending queue");
+
+            // If the entry for the client ID already exists (in case of subsequent ICE candidates), update the queue
+            if(pendingIceCandidatesMap.containsKey(message.getSenderClientId())) {
+                Queue<IceCandidate> pendingIceCandidatesQueueByClientId = pendingIceCandidatesMap.get(message.getSenderClientId());
+                pendingIceCandidatesQueueByClientId.add(iceCandidate);
+                pendingIceCandidatesMap.replace(message.getSenderClientId(), pendingIceCandidatesQueueByClientId);
```

Comment:
Why this line? No need to replace antyhing

Reply:
Actually, you are right. I added this line when I thought `replace` was different from `put`. Will fix it.